### PR TITLE
Added logout to dropdown menu

### PR DESCRIPTION
**Pull request by koyuawsmbrtn**
### Description

First bigger pull request. ~~Need some review on localization, haven't tested this PR yet.~~

This will make logging out off Mastodon a little bit easier to new users.

### Preview

image

### Task list

- [x] Creating pull request
- [x] Test and fix

### CODE REVIEW

File: app/javascript/mastodon/components/dropdown_menu.js [hunk 1]
```diff
@@ -124,9 +124,16 @@ class DropdownMenu extends React.PureComponent {
 
     const { text, href = '#' } = option;
```
Comment (Gargron) [inline app/javascript/mastodon/components/dropdown_menu.js]: Add `method`

File: app/javascript/mastodon/components/dropdown_menu.js [hunk 2]
```diff
@@ -124,9 +124,16 @@ class DropdownMenu extends React.PureComponent {
...
 
+    var target = '_blank';
+    var method = '';
+    if (href === '/auth/sign_out') {
```
Comment (Gargron) [inline app/javascript/mastodon/components/dropdown_menu.js]: Remove this section and just pass `method` from outside

File: app/javascript/mastodon/features/compose/components/action_bar.js [hunk 3]
```diff
@@ -42,6 +43,8 @@ class ActionBar extends React.PureComponent {
     menu.push({ text: intl.formatMessage(messages.blocks), to: '/blocks' });
     menu.push({ text: intl.formatMessage(messages.domain_blocks), to: '/domain_blocks' });
     menu.push({ text: intl.formatMessage(messages.filters), href: '/filters' });
+    menu.push(null);
+    menu.push({ text: intl.formatMessage(messages.logout), href: '/auth/sign_out' });
```
Comment (Gargron) [inline app/javascript/mastodon/features/compose/components/action_bar.js]: I.e. add `method: 'delete'` here

**Review by Gargron — CHANGES_REQUESTED** [review]

File: app/javascript/mastodon/features/compose/components/action_bar.js [hunk 4]
```diff
@@ -44,7 +44,7 @@ class ActionBar extends React.PureComponent {
     menu.push({ text: intl.formatMessage(messages.domain_blocks), to: '/domain_blocks' });
     menu.push({ text: intl.formatMessage(messages.filters), href: '/filters' });
     menu.push(null);
-    menu.push({ text: intl.formatMessage(messages.logout), href: '/auth/sign_out' });
+    menu.push({ text: intl.formatMessage(messages.logout), href: '/auth/sign_out', target: '', method: 'delete' });
```
Comment (Gargron) [inline app/javascript/mastodon/features/compose/components/action_bar.js]: If you pass `null` the `target` attribute won't be rendered.

File: app/javascript/mastodon/components/dropdown_menu.js [hunk 5]
```diff
@@ -122,14 +122,7 @@ class DropdownMenu extends React.PureComponent {
...
 
-    const { text, href = '#' } = option;
-
-    var target = '_blank';
-    var method = '';
-    if (href === '/auth/sign_out') {
-      target = '';
-      method = 'delete';
-    }
+    const { text, href = '#', target = '_blank', method = '' } = option;
```
Comment (Gargron) [inline app/javascript/mastodon/components/dropdown_menu.js]: There is no need to set `method` to any default value. We don't want it rendered when nothing is given.

**Review by Gargron — CHANGES_REQUESTED** [review]

**Comment by koyuawsmbrtn** [comment]
Hope this works now.

**Review by Gargron — APPROVED** [review]

**Pull request merged**